File: app/src/main/java/net/ducksmanager/persistence/models/composite/UserSetting.java
package net.ducksmanager.persistence.models.composite;

import org.jetbrains.annotations.NotNull;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "user_settings")
public class UserSetting {

    @PrimaryKey
    @NonNull
    private String settingKey;

    @ColumnInfo
    private String value;

    public UserSetting(@NotNull String settingKey, String value) {
        this.settingKey = settingKey;
        this.value = value;
    }

    @NotNull
    public String getSettingKey() {
        return settingKey;
    }

    public void setSettingKey(@NotNull String settingKey) {
        this.settingKey = settingKey;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }
}

File: app/src/main/java/net/ducksmanager/persistence/models/composite/UserSetting.kt
package net.ducksmanager.persistence.models.composite

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import org.jetbrains.annotations.NotNull

@Entity(tableName = "user_settings")
class UserSetting(settingKey: @NotNull String?, value: String) {

    companion object {
        const val SETTING_KEY_NOTIFICATIONS_ENABLED = "SETTING_KEY_NOTIFICATIONS_ENABLED"
        const val SETTING_KEY_ISSUE_SELECTION_TIP_ENABLED = "SETTING_KEY_ISSUE_SELECTION_TIP_ENABLED"
    }

    @PrimaryKey
    @NonNull
    private var settingKey: String

    @ColumnInfo
    var value: String

    fun getSettingKey(): @NotNull String = settingKey

    fun setSettingKey(settingKey: @NotNull String?) {
        this.settingKey = settingKey!!
    }

    init {
        this.settingKey = settingKey!!
        this.value = value
    }
}